packages/typescript-private/device-server/src/internal/externals/cli/webdriver-agent-process.ts

```diff
@@ -0,0 +1,204 @@
+import { HostPaths } from '@dogu-tech/node';
+import axios, { AxiosInstance } from 'axios';
+import http from 'http';
+import { Zombieable, ZombieProps, ZombieWaiter } from '../../services/zombie/zombie-component';
+import { ZombieServiceInstance } from '../../services/zombie/zombie-service';
+import { XcodeBuild } from '../index';
+import { DerivedData } from '../xcode/deriveddata';
+import { MobileDevice } from './mobiledevice';
+import { ZombieTunnel } from './mobiledevice-tunnel';
+import { XCTestRunContext } from './xcodebuild';
+
+export class WebdriverAgentProcess {
+  private readonly xctest: ZombieWdaXCTest;
+  private readonly wdaTunnel: ZombieTunnel;
+
+  constructor(private readonly serial: Serial, private readonly wdaHostPort: number, private readonly logger: Printable, private isKilled = false) {
+    ZombieServiceInstance.deleteComponentIfExist((zombieable: Zombieable): boolean => {
+      if (zombieable instanceof ZombieWdaXCTest) {
+        return zombieable.serial === this.serial;
+      }
+      return false;
+    }, 'kill previous ZombieWdaXCTest');
+    ZombieServiceInstance.deleteComponentIfExist((zombieable: Zombieable): boolean => {
+      if (zombieable instanceof ZombieTunnel) {
+        return zombieable.serial === this.serial && zombieable.hostPort === this.wdaHostPort;
+      }
+      return false;
+    }, 'kill previous tunnel');
+    this.xctest = new ZombieWdaXCTest(this.serial, wdaHostPort, this.logger);
+    this.wdaTunnel = new ZombieTunnel(this.serial, wdaHostPort, 8100, this.logger);
+  }
+
+  static async start(serial: Serial, wdaHostPort: number, logger: Printable): Promise<WebdriverAgentProcess> {
+    const originDerivedData = await DerivedData.create(HostPaths.external.xcodeProject.wdaDerivedDataPath());
+    if (!originDerivedData.hasSerial(serial)) {
+      throw new Error(`WebdriverAgent can't be executed on ${serial}`);
+    }
+    const ret = new WebdriverAgentProcess(serial, wdaHostPort, logger);
+    await ret.xctest.zombieWaiter.waitUntilAlive();
+    await ret.wdaTunnel.zombieWaiter.waitUntilAlive();
+    return ret;
+  }
+
+  delete(): void {
+    ZombieServiceInstance.deleteComponent(this.xctest);
+    ZombieServiceInstance.deleteComponent(this.wdaTunnel);
+  }
+}
+
+class ZombieWdaXCTest implements Zombieable {
+  private xctestrun: XCTestRunContext | null = null;
+  public readonly zombieWaiter: ZombieWaiter;
+  private error: 'not-alive' | 'hello-failed' | 'none' = 'none';
+  private client: AxiosInstance;
+
+  constructor(public readonly serial: Serial, private readonly wdaHostPort: number, private readonly logger: Printable) {
+    this.zombieWaiter = ZombieServiceInstance.addComponent(this);
+    this.client = axios.create({
+      baseURL: `http://127.0.0.1:${wdaHostPort}`,
+      httpAgent: new http.Agent({ keepAlive: true, maxSockets: 1, maxTotalSockets: 1, maxFreeSockets: 0 }),
+      timeout: 5000,
+      maxRedirects: 10,
+      maxContentLength: 50 * 1000 * 1000,
+    });
```

Comment:
please add setAxiosErrorFilterToIntercepter(this.client) for long long exios error log